p2p/pex/pex_reactor.go

```diff
@@ -560,7 +542,8 @@ func (r *Reactor) dialPeer(addr *p2p.NetAddress) error {
 
 	err := r.Switch.DialPeerWithAddress(addr)
 	if err != nil {
-		if _, ok := err.(p2p.ErrCurrentlyDialingOrExistingAddress); ok {
+		var errCurrentlyDialingOrExistingAddress p2p.ErrCurrentlyDialingOrExistingAddress
+		if errors.As(err, &errCurrentlyDialingOrExistingAddress) {
```

Comment:
Not really pleasant this method for checking the type of an error.

Reply:
Done.